src/app/services/article.service.ts

```diff
@@ -73,6 +79,51 @@ export class ArticleService {
     return this.articles;
   };
 
+  public fetchAllArticlesFromFirestore(): Observable<Article[]> {
+    const url = `${this.BASE_FIRESTORE}/projects/auxilium-420904/databases/aux-db/documents:runQuery`;
+    const headers: HttpHeaders = new HttpHeaders({'Content-Type': 'application/json'});
+    const body = {
+      structuredQuery: {
+        from: [{collectionId: 'articles'}],
+      }
+    };
+
+    return this.http.post(url, body, {headers}).pipe(
+      map((documents: any): Article[] => {
+
+        const fetchedArticles: Article[] = [];
+        documents.forEach((document: any) => {
+          const fields = document.document.fields;
+          let documentId: string = '';
+
+          /* Todo: This is bad. We need our database to adhere to the contract of data blobs that we expect.
```

Comment:
I created a script last night to fix the `header` (formerly `title`), `imageURI` issue. All should have a `body` even before.

I added `https://t4.ftcdn.net/jpg/02/97/78/03/360_F_297780357_pK8VCA7wctbTFusAGiCfcoxbJLRwC9Bs.jpg` as a placeholder image and set `imageURI` with it, but I don't think it needs a value if it doesn't have an image.

`imageURI` should be optional. The default image can be added at runtime.

Reply:
I suppose that's what `line 105` does below.